services/venice-server/src/main/java/com/linkedin/venice/listener/OutboundHttpWrapperHandler.java

```diff
@@ -162,4 +169,65 @@ public void write(ChannelHandlerContext ctx, Object msg, ChannelPromise promise)
      */
     ctx.writeAndFlush(response);
   }
+
+  @Override
+  public void grpcRead(GrpcHandlerContext ctx, GrpcHandlerPipeline pipeline) {
+    pipeline.processRequest(ctx);
+  }
+
+  @Override
+  public void grpcWrite(GrpcHandlerContext ctx, GrpcHandlerPipeline pipeline) {
+
+    ByteBuf body;
+    CompressionStrategy compressionStrategy = CompressionStrategy.NO_OP;
+
+    ReadResponse obj = ctx.getReadResponse();
+    GrpcStatsContext statsContext = ctx.getGrpcStatsContext();
+    VeniceServerResponse.Builder veniceServerResponseBuilder = ctx.getVeniceServerResponseBuilder();
+    if (!ctx.hasError()) {
+      compressionStrategy = obj.getCompressionStrategy();
+
+      statsContext.setDatabaseLookupLatency(obj.getDatabaseLookupLatency());
```

Comment:
Hmm... the stuff here seems copy-pasted... can you extract common pieces of code into functions that will be called from both paths?